comp/core/workloadmeta/collectors/internal/kubeapiserver/kubeapiserver.go

```diff
@@ -106,12 +107,12 @@ func resourcesWithExplicitMetadataCollectionEnabled(cfg config.Reader) []string
 	var resources []string
 	requestedResources := cfg.GetStringSlice("cluster_agent.kube_metadata_collection.resources")
 	for _, resource := range requestedResources {
-		if resource == "pods" && shouldHavePodStore(cfg) {
+		if strings.HasSuffix(resource, "pods") && shouldHavePodStore(cfg) {
```

Comment:
this will be a false positive if the user uses a resource named `pod` that is not the native kubernetes pod. 
However, handling this extreme edge case complicates the code a lot, and this configuration option should not normally exposed to the user via helm/operator, but rather used by other agent components to collect useful metadata.

Reply:
We can try to solve this with a regex in a future PR.

Reply:
I don't think it is only about using regex. 

It is about defining what we classify as native deployment (or native pod).

If we are using regex, we will end up having something like `^[^\/]+\/([^\/]*\/)?deployments$`, which is basically almost the same as checking only the suffix, because at the end we will parse the string when finding the GVR and skip it if it doesn't follow the correct format. 

For example, if we have `apps/v2/deployment`, is this a native deployment? How can we know (even with a regex)? 
It might be that some user decided to create this custom resource (which as mentioned, is an extreme edge case). 

Similarly, what if the group of pods changes to a new group (which is also highly unlikely, but not impossible), wouldn't that break (even if we are using regex)?

Reply:
I misunderstood the problem at first. It's clear now :+1: